source/validate_id.cpp

```diff
@@ -2377,6 +2382,29 @@ function<bool(unsigned)> getCanBeForwardDeclaredFunction(SpvOp opcode) {
 
 namespace libspirv {
 
+spv_result_t UpdateIdUse(ValidationState_t& _) {
+  for (const auto& inst : _.ordered_instructions()) {
+    for (auto& operand : inst.operands()) {
+      const spv_operand_type_t& type = operand.type;
+      const uint32_t operand_id = inst.words()[operand.offset];
+
+      switch (type) {
+        case SPV_OPERAND_TYPE_VARIABLE_ID:
```

Comment:
Please use spvIsIdType(type) from operand.h instead.
There are already a few places where we see the same 5 type ids in a row, and that's a bit painful, but they're all in switch statements with other cases that need to be addressed.